src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -15,7 +19,27 @@ public void setup() {
     }
 
     @Test
-    public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+    public void testFindProductsStart() {
+        assertFalse(productDao.findProducts().isEmpty());
+    }
+
+    @Test
+    public void testSaveNewProduct() {
+
+        Currency usd = Currency.getInstance("USD");
+        Product testProduct = new Product("sgs", "Samsung Galaxy S", new BigDecimal(100), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
+
+        productDao.save(testProduct);
+
+        assertTrue(testProduct.getId() > 0);
+        assertNotNull(productDao.getProduct(testProduct.getId()));
+        assertSame(productDao.getProduct(testProduct.getId()), testProduct);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void testDeleteProduct() {
+        Long id = 1L;
+        productDao.delete(id);
+        productDao.getProduct(id);
```

Comment:
Please add more tests for each aspect of functionality: get, find, save, delete product.
Add more testing for corner cases: what if we search for null product? What is we try to save the product without ID? Delete product that does not exist? etc.

Reply:
done